src/adapters/deno.js

```diff
@@ -19,24 +17,35 @@ async function fileFrom (path) {
 export class Sink {
   /**
    * @param {Deno.File} fileHandle
+   * @param {string} path
    * @param {number} size
    */
-  constructor (fileHandle, size) {
+  constructor (fileHandle, path, size) {
     this.fileHandle = fileHandle
+    this.path = path
     this.size = size
     this.position = 0
   }
   async abort() {
-    await this.fileHandle.close()
+    this.fileHandle.close()
   }
   async write (chunk) {
+    try {
+      await Deno.stat(this.path)
+    } catch (err) {
+      if (err.name === 'NotFound') {
+        this.fileHandle.close()
+        throw new DOMException(...GONE)
+      }
+    }
```

Comment:
note to self, it's to expensive to do this operation on every chunk-write 
it should only stat the file before creating a sink.

Reply:
I remember adding this because a test was failing, most likely `write() fails when parent directory is removed`. It's not enough to check when creating the sink and on `close()`, but also on each `write`. That's the native behavior, although I couldn't find this documented anywhere. Also a bit odd because the file is not written to in-place, but the actual write happens at the end on `close()`.

Reply:
yea... they write to some temporary file and then they move/replace it...

Reply:
Also i learned that this works out totally fine in chrome:

```js
root = await navigator.storage.getDirectory()
handle = await root.getFileHandle('abc', {create: true})
await root.removeEntry('abc')
wr = await handle.createWritable()
```

I asked them if it was intentional https://github.com/WICG/file-system-access/issues/334